docs/tasks/access-kubernetes-api/migrate-third-party-resource.md

```diff
@@ -0,0 +1,163 @@
+start from scratch with a CRD.
+This page describes an optional process that eases the transition by migrating existing TPR data for
+you **on a best-effort basis**.
+{% endcapture %}
+
+{% capture prerequisites %}
+* Make sure your Kubernetes cluster has a **master version of exactly 1.7.x** (any patch release),
+  as this is the only version that supports both TPR and CRD.
+* If you use a TPR-based custom controller, check with the author of the controller first.
+  Some or all of these steps may be unnecessary if the custom controller handles the migration for
+  you.
+* Be familiar with the concept of [custom resources](/docs/concepts/extensions/custom-resources/),
+  which were known as *third-party resources* until Kubernetes 1.7.
+* Be familiar with [how to use CustomResourceDefinitions](/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/).
+* **Before performing a migration on real data, conduct a dry run of these steps in a test cluster.**
+{% endcapture %}
+
+{% capture steps %}
+## Migrate TPR data
+
+1.  **Rewrite the TPR definition**
+
+    Clients that access the REST API for your custom resource should not need any changes.
+    However, you will need to rewrite your TPR definition as a CRD.
+
+    Make sure you specify values for the CRD fields that match what the server used to fill in for
+    you with TPR.
+
+    For example, if your ThirdPartyResource looks like this:
+
+    ```yaml
+    apiVersion: extensions/v1beta1
+    kind: ThirdPartyResource
+    metadata:
+      name: cron-tab.stable.example.com
+    description: "A specification of a Pod to run on a cron style schedule"
+    versions:
+    - name: v1
+    ```
+
+    A matching CustomResourceDefinition could look like this:
+
+    ```yaml
+    apiVersion: apiextensions.k8s.io/v1beta1
+    kind: CustomResourceDefinition
+    metadata:
+      name: crontabs.stable.example.com
+    spec:
+      scope: Namespaced
+      group: stable.example.com
+      version: v1
+      names:
+        kind: CronTab
+        plural: crontabs
+        singular: crontab
+    ```
+
+1.  **Install the CustomResourceDefinition**
+
+    While the source TPR is still active, install the matching CRD with `kubectl create`.
+    Existing TPR data will remain accessible because TPRs take precedence over CRDs when both try
+    to serve the same resource.
+
+    After you create the CRD, make sure the *Established* condition goes to True.
+    You can check it with a command like this:
+
+    ```shell
+    kubectl get crd -o 'custom-columns=NAME:{.metadata.name},ESTABLISHED:{.status.conditions[?(@.type=="Established")].status}'
+    ```
+
+    The output should look like this:
+
+    ```console
+    NAME                          ESTABLISHED
+    crontabs.stable.example.com   True
+    ```
+
+1.  **Stop all clients that use the TPR**
+
+    The API server will attempt to prevent mutation of TPR data for the resource while it
```

Comment:
[RW] The API server attempts to prevent TPR data for the resource from mutation while it...
avoids future tense and nominalization (easier to read)